src/interaction/FixedLocalTupleComListInteractionTemplate.hpp

```diff
@@ -0,0 +1,408 @@
+  
+  This file is part of ESPResSo++.
+  
+  ESPResSo++ is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  
+  ESPResSo++ is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+*/
+
+// ESPP_CLASS
+#ifndef _INTERACTION_FIXEDLOCALTUPLECOMLISTINTERACTIONTEMPLATE_HPP
+#define _INTERACTION_FIXEDLOCALTUPLECOMLISTINTERACTIONTEMPLATE_HPP
+
+#include "mpi.hpp"
+#include "Interaction.hpp"
+#include "Real3D.hpp"
+#include "Tensor.hpp"
+#include "Particle.hpp"
+#include "FixedLocalTupleList.hpp"
+#include "esutil/Array2D.hpp"
+#include "bc/BC.hpp"
+#include "SystemAccess.hpp"
+#include "storage/Storage.hpp"
+#include "Interaction.hpp"
+#include "types.hpp"
+#include "esutil/Error.hpp"
+
+namespace espressopp {
+    namespace interaction {
+	template < typename _Potential >
+	class FixedLocalTupleComListInteractionTemplate: public Interaction, SystemAccess {
+	    
+	protected:
+	    typedef _Potential Potential;
+	    
+	private:
+	    long unsigned int num_of_part; // total number of particles
+	    long unsigned int num_of_subchain; // total number of subchains
+	    std::map<long unsigned int, Real3D> com_origin;
+	    
+	    int N_Constrain; // number of particles constraining the interaction
+	    int nParticles;  // number of particles in this node
+	    
+	public:
+	    FixedLocalTupleComListInteractionTemplate
+	    (shared_ptr < System > _system,
+	     shared_ptr < FixedLocalTupleList > _fixedtupleList,
+	     shared_ptr < Potential > _potential)
+	      : SystemAccess(_system), fixedtupleList(_fixedtupleList), potential(_potential) {
+		if (! potential) {
+		    LOG4ESPP_ERROR(theLogger, "NULL potential");
+		}
+		System& system = getSystemRef();
+		esutil::Error err(system.comm);
+
+		long unsigned int localN = system.storage -> getNRealParticles();
+		boost::mpi::all_reduce(*system.comm, localN, num_of_part, std::plus<int>());
+
+		FixedLocalTupleList::TupleList::Iterator it(*fixedtupleList);
+		std::vector<Particle*> pList = it->second;
+
+		N_Constrain = pList.size() + 1;
+
+		// Check the length of individula particle list in FixedLocalTuple
+		for (FixedLocalTupleList::TupleList::Iterator it(*fixedtupleList); it.isValid(); ++it) {
+		    std::vector<Particle*> pList = it->second;
+		    
+		    if (pList.size() + 1 != N_Constrain) {
+			std::stringstream msg;
+			msg << "ERROR: Tuple Length is not constant\n";
+			err.setException(msg.str());
+		    }
+		}
+		
+		num_of_subchain = num_of_part/N_Constrain;
+
```

Comment:
This class must know the desired center of mass (COM) of all subchains for constraining the COM of individual subchain. Hence, I need to call all_reduce and set `Real3D *totCOM = new Real3D[num_of_subchain]` and `Real3D *COM = new Real3D[num_of_subchain]`.
In contrast, temporal COM of subchain is implemented using `unordered_map`.